src/nvim/os/signal.c

```diff
@@ -158,24 +158,18 @@ static void on_signal(SignalWatcher *handle, int signum, void *data)
   }
 }
 
-static void signal_unblock_SIGCHLD(void)
+static void signal_unblock(int signo)
 {
 #ifndef WIN32
   sigset_t mask;
   sigemptyset(&mask);
 
-  // Work around broken macOS headers. #5243
-# if defined(__APPLE__) && !defined(__clang__) && !defined(__INTEL_COMPILER) \
-    && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
-#  pragma GCC diagnostic push
-#  pragma GCC diagnostic ignored "-Wsign-conversion"
-# endif
-
-  sigaddset(&mask, SIGCHLD);
-
-# if defined(__APPLE__) && !defined(__clang__) && !defined(__INTEL_COMPILER) \
-    && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
-#  pragma GCC diagnostic pop
+  // macOS has type-unsafe inline functions and macro's in signal.h, see
+  // #5243. This will start warning when Apple fixes their headers.
+# if defined(__APPLE__) && defined(__GNUC__) && !defined(__clang__)
```

Comment:
> There's one possible edge case, if GCC ever decides to stop spewing warnings for macro's from system headers like Clang, we would be none the wiser.

How would

    # if defined(__APPLE__) && defined(__GNUC__) && !defined(__clang__)

differ from

    # if defined(__APPLE__)

in that case?